### PR TITLE
AA-711: implement upsell click events

### PR DESCRIPTION
**Pull request by ciduarte**
Adding `course_home_green` and `in_course_upgrade` upsell events to course home and courseware.

Descriptions of these events:
https://openedx.atlassian.net/wiki/spaces/RS/pages/1675100377/How+to+find+upsell+link+click+events

### CODE REVIEW

File: src/courseware/course/sequence/lock-paywall/LockPaywall.jsx [hunk 1]
```diff
@@ -25,6 +27,20 @@ function LockPaywall({
...
+
+  const eventProperties = {
+    org_key: org,
+    courserun_key: courseId,
+  };
+
+  const logClick = () => {
+    sendTrackingLogEvent('edx.bi.ecommerce.upsell_links_clicked', {
+      ...eventProperties,
+      linkCategory: '(none)',
```
Comment (ciduarte) [inline src/courseware/course/sequence/lock-paywall/LockPaywall.jsx:39]: So, I know this looks weird, but I'm under the impression that `'(none)'` is what they want here... I'll double check w/ rev

See:
https://github.com/edx/edx-platform/pull/24526#discussion_r459447956

https://openedx.atlassian.net/wiki/spaces/RS/pages/1675100377/How+to+find+upsell+link+click+events

File: src/course-home/outline-tab/widgets/UpgradeCard.jsx [hunk 2]
```diff
@@ -46,6 +46,12 @@ function UpgradeCard({ courseId, intl, onLearnMore }) {
       location: 'sidebar-message',
     });
     sendTrackEvent('Promotion Clicked', promotionEventProperties);
+    sendTrackingLogEvent('edx.bi.ecommerce.upsell_links_clicked', {
```
Comment (MatthewPiatetsky) [inline src/course-home/outline-tab/widgets/UpgradeCard.jsx]: what is the difference between sendTrackEvent and sendTrackingLogEvent, first goes to segment and second to splunk?

**Review by MatthewPiatetsky — APPROVED** [review]
a review from someone on rev squad familiar with these events might be a good idea

**Review by mikix — APPROVED** [review]

File: src/courseware/course/sequence/lock-paywall/LockPaywall.jsx (hunk 1 above)
Comment (dianekaplan) [reply src/courseware/course/sequence/lock-paywall/LockPaywall.jsx:39]: Yes sorry- this awkward choice was my fault, you're right that this is the way to keep it consistent

**Review by dianekaplan — CHANGES_REQUESTED** [review]
Excited to see this started so soon!

One general flag: I don't see the actual link names passed along to sendTrackEvent or sendTrackingLogEvent in the code (course_home_green and in_course_upgrade)- not sure which one sends to Segment? 

Our trackUpsellClick function in edx-platform took the link name along with the set of properties (pageName, linkType, linkCategory), so segment events have `event:properties:linkName` that we search on, also used by our New Relic graph to see the relative counts of the different upsell links. In this code I don't see the link names being passed along- is this happening in a subtle way that I'm missing, or still to be done?

**Comment by ciduarte** [comment]
> One general flag: I don't see the actual link names passed along to sendTrackEvent or sendTrackingLogEvent in the code (course_home_green and in_course_upgrade)- not sure which one sends to Segment?
> 
> Our trackUpsellClick function in edx-platform took the link name along with the set of properties (pageName, linkType, linkCategory), so segment events have `event:properties:linkName` that we search on, also used by our New Relic graph to see the relative counts of the different upsell links. In this code I don't see the link names being passed along- is this happening in a subtle way that I'm missing, or still to be done?

The `linkName` is definitely something I missed! Glad you caught that.

**Comment by ciduarte** [comment]
>  not sure which one sends to Segment?

`sendTrackEvent` goes to Segment, so that's what we actually want. Updated it to use that rather than `sendTrackingLogEvent`.

**Review by dianekaplan — APPROVED** [review]
Thanks for adding the linkName! No other feedback for these changes, so once you've got the new sendTrackEvent lines under test this will be good to go. Thanks!

**Pull request merged**